---
title: "elt2_RNAseq"
author: "Robert Williams"
date: "3/9/2022"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load packages
```{r load-packages, include=FALSE}
library(DESeq2)
library(corrplot)
library(RColorBrewer)
library(pheatmap)
library(tidyverse)
library(apeglm)
library(ashr)
library(InterMineR)
library(ComplexHeatmap)
```

Load count matrix

```{r}
elt2D_counts <- read_csv(file = "../01_input/Table_S1_Raw_Read_Counts_wt_elt2D.csv") %>% column_to_rownames(var = "WBGeneID")
head(elt2D_counts)
```

```{r}
coldata <- data.frame(condition = c(rep("wt", 4), rep("elt2D", 4)))
coldata$condition <- factor(coldata$condition, levels = c("wt","elt2D"))
rownames(coldata) <- colnames(elt2D_counts)
all(rownames(coldata) == colnames(elt2D_counts))
```

```{r}
dds_elt2 <- DESeqDataSetFromMatrix(countData = elt2D_counts,
                              colData = coldata,
                              design = ~ condition)
```

Visualize read count distribution
```{r}
raw_count_threshold <- 10
hist(log10(rowSums(counts(dds_elt2))), breaks = 50)
abline(v = log10(raw_count_threshold), col = "red", lty = 2)
```
# Filter genes with sum counts per million >= 10 across all samples

```{r}
cpm <- apply(counts(dds_elt2),2, function(x) (x/sum(x))*1000000)
hist(log10(rowSums(cpm)))
abline(v = log10(raw_count_threshold), col = "red", lty = 2)
```
Filter genes with low read counts
```{r}
keep <- rowSums(cpm) >= raw_count_threshold
dds_elt2 <- dds_elt2[keep,]
dds_elt2
```

Perform Differential Expression
```{r}
dds_elt2 <- DESeq(dds_elt2)
resultsNames(dds_elt2)
```

# Sample-to-sample distance matrix

```{r}
vsd <- vst(dds_elt2, blind = FALSE)
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- colnames(vsd)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```
# Differential expression

```{r}
res_elt2D_v_wt <- results(dds_elt2)
```


```{r}
plotMA(dds_elt2, ylim = c(-10,10))
```
Output data table

```{r}
# write_csv(res_to_df(res_elt2D_v_wt), file = "../03_output/res_elt2D_v_wt.csv")
```


# Log2Shrinkage for gene ranking

```{r}
res_elt2D_v_wt_ashr <- lfcShrink(dds_elt2, coef = "condition_elt2D_vs_wt", type = "ashr")
# write_csv(file = "../03_output/res_elt2D_v_wt_ashr_shrunk.csv", x = res_to_df(res_elt2D_v_wt_ashr))
```

```{r}
plotMA(res_elt2D_v_wt_ashr, ylim = c(-10,10))
```

# Alternative hypothesis testing

```{r}
thresh <- 1
sig <- 0.01
res_altHyp_greater <- results(dds_elt2, altHypothesis = "greater", lfcThreshold = thresh, alpha = sig)
res_altHyp_less <- results(dds_elt2, altHypothesis = "less", lfcThreshold = thresh, alpha = sig)
res_altHyp_lessAbs <- results(dds_elt2, altHypothesis = "lessAbs", lfcThreshold = thresh, alpha = sig)
```


```{r}
par(mfrow=c(1,3),mar=c(2,2,1,1))
ylim <- c(-10,10)
plotMA(res_altHyp_greater, ylim = ylim)
plotMA(res_altHyp_less, ylim = ylim)
plotMA(res_altHyp_lessAbs, ylim = ylim)
```

```{r}
elt2D_greater <- as.data.frame(res_altHyp_greater) %>% 
  rownames_to_column(var = "WBGeneID") %>% 
  filter(padj < 0.01) %>% mutate(altHyp = "greater", description = "up_ELT2_minus")

elt2D_less <- as.data.frame(res_altHyp_less) %>% 
  rownames_to_column(var = "WBGeneID") %>% 
  filter(padj < 0.01) %>% mutate(altHyp = "less", description = "down_ELT2_minus")

elt2D_lessAbs <- as.data.frame(res_altHyp_lessAbs) %>% 
  rownames_to_column(var = "WBGeneID") %>% 
  filter(padj < 0.01) %>% mutate(altHyp = "lessAbs", description = "unchanged_ELT2_minus")

elt2_regulated_genes <- bind_rows(elt2D_greater, elt2D_less, elt2D_lessAbs) %>% select(WBGeneID, altHyp, description)
elt2_regulated_genes$description <- factor(elt2_regulated_genes$description, levels = c("up_ELT2_minus", "down_ELT2_minus", "unchanged_ELT2_minus"))
head(elt2_regulated_genes)
```
Add gene names

```{r}

paramart <- biomaRt::useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

elt2_regulated_genes <- biomaRt::getBM(
  mart = paramart,
  filter = c("wbps_gene_id"),
  value = elt2_regulated_genes$WBGeneID,
  attributes = c("wbps_gene_id","wormbase_gseq", "wikigene_name")
) %>% right_join(elt2_regulated_genes, by = c("wbps_gene_id" = "WBGeneID"))

elt2_regulated_genes<- elt2_regulated_genes %>% rename(WBGeneID = "wbps_gene_id")

# write_csv(elt2_regulated_genes, file = "../03_output/elt2_regulated_gene_sets.csv")
```


Counts of genes in each category

```{r}
elt2_regulated_gene_counts <- elt2_regulated_genes %>% group_by(description) %>% summarise(genes = n()) %>% #fct_recode(description = c("")) %>%
  ggplot(aes(x = description, y = genes, fill = description, label = genes)) +
  geom_bar(stat = "identity") +
  geom_text(vjust = -0.5) +
  ylim(c(0, 10000)) +
  coord_flip() +
  theme_bw()

elt2_regulated_gene_counts

# ggsave(elt2_regulated_gene_counts, filename = "../03_output/elt2_regulated_gene_counts_plot.pdf", width = 6, height = 3, useDingbats = FALSE)
```

# MA plot

```{r}
res_elt2D_v_wt_ashr_altHyp <- as.data.frame(res_elt2D_v_wt_ashr) %>% rownames_to_column(var = "WBGeneID") %>% left_join(elt2_regulated_genes, by = "WBGeneID")

elt2D_vs_wt_MA <- ggplot(res_elt2D_v_wt_ashr_altHyp %>% filter(!is.na(description)), aes(x = log10(baseMean), y = log2FoldChange, color = description)) +
  geom_point(data = res_elt2D_v_wt_ashr_altHyp, shape = 20, alpha = 0.5, stroke = 0, size = 2, color = "grey") +
  geom_point(shape = 20, alpha = 0.25, stroke = 0, size = 2) +
  theme_bw()

elt2D_vs_wt_MA

ggsave(filename = "../03_output/elt2D_vs_wt_MA_plot.pdf",elt2D_vs_wt_MA, width = 5, height = 3, useDingbats=FALSE)
```

# Volcano plot

```{r}
elt2D_vs_wt_volcano <- ggplot(res_elt2D_v_wt_ashr_altHyp %>% filter(padj !=0, !is.na(description)), 
       aes(x = log2FoldChange, y = -log10(padj), color = description)) +
  geom_point(data = res_elt2D_v_wt_ashr_altHyp %>% filter(padj !=0) %>% dplyr::select(-description),shape = 20, alpha = 0.75, stroke = 0, size = 2, color = "darkgrey") +
  geom_text(data = res_elt2D_v_wt_ashr_altHyp %>% filter(padj !=0, !is.na(description)) %>% slice_max(-log10(padj), n= 5), aes(x = log2FoldChange, y = -log10(padj), label = wikigene_name)) +
  geom_point(shape = 20, alpha = 0.75, stroke = 0, size = 2) +
  theme_bw()

elt2D_vs_wt_volcano

ggsave(filename = "../03_output/elt2D_vs_wt_volcano_plot.pdf", elt2D_vs_wt_volcano, width = 5, height = 3, useDingbats = FALSE)
```


# Session info

```{r}
sessionInfo()
```
