---
title: "embryo_L1_L3_intestine_RNAseq"
author: "Rtpw"
date: "2/8/2022"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Install libraries
```{r}

```

Load libraries
```{r load-packages, include=FALSE}
library(DESeq2)
library(corrplot)
library(RColorBrewer)
library(pheatmap)
library(tidyverse)
library(apeglm)
library(ashr)
library(InterMineR)
library(ComplexHeatmap)
```

```{r source-functions}
source(file = "../../elt2_paper_functions.R")
```


# Analysis outline

- Import counts and format for DESeq2 analysis
- Filter low abundance counts based on counts per million (CPM)
- Filter for protein coding genes
- Perform differential expression analysis
- Visualize pairwise sample correlation
- Categorize gene expression with alternative hypothesis
- Visualize unshrunken pairwise differential expression
- Visualize shrunken pairwise differential expression
- Visualize developmentally dynamic intestine genes
- Output data tables

# Import and format count data

```{r}
# import counts
countsData <- read.delim(file = "../01_input/all.counts", sep = " ")
# preview counts
head(countsData)
# print samples
colnames(countsData[6:ncol(countsData)])
# import metadata and process file
metadata1 <- read.table(file = "../01_input/RWP27_metadata.tsv", header = FALSE, stringsAsFactors = FALSE) %>% bind_rows(read.table(file = "../01_input/RWP26_metadata.tsv", header = FALSE, stringsAsFactors = FALSE)) %>%
  bind_rows(read.table(file = "../01_input/RWP30_metadata.tsv", header = FALSE, stringsAsFactors = FALSE))

colnames(metadata1) <- c("Filename.Fwd", "Filename.Rev", "names")
head(metadata1)

# separate and process sample info
metadata1 <- metadata1 %>% separate(names, sep = "_", into = c("stage", "sample", "rep"), remove = FALSE)
metadata1 <- metadata1 %>% mutate(stage = fct_relevel(stage, c("embryo", "L1", "L3")), 
                     sample = fct_relevel(sample, c("whole", "cells", "GFPplus", "GFPminus")),
                     rep = fct_relevel(rep, c("rep1", "rep2", "rep3")),
                     names = fct_relevel(names, metadata1$names)
                     )

# Order columns according to metadata1 order
countsData <- countsData  %>% select(chr:length, sort(metadata1$names))
head(countsData)

# Generate a table called "cts" out of the countsData table. Subset the countsData.
cts <- as.matrix(countsData %>% select(metadata1$names))
head(cts)

# Reorganize the metadata table so the names2 column are now headers
rownames(metadata1)<- metadata1$names
coldata <- metadata1[,c("names", "stage", "sample", "rep")]
rownames(coldata) <- as.vector(metadata1$names)
# make grouping variable
coldata$group <- factor(paste0(coldata$stage, coldata$sample))
coldata

# Check that the names match  --> Should be TRUE
all(rownames(coldata) == colnames(cts))

```


# Make DESeqDataSet

Generate the DESeqDataSet. The variables in this design formula will be the type of sample, and the preparation date. This should reduce the variability between the samples based on when they were made.

From the vignette: "In order to benefit from the default settings of the package, you should put the variable of interest at the end of the formula and make sure the control level is the first level."

The variable of interest is the sample `type`.

Using `DESeqDataSetFromMatrix` since I used the program `featureCounts`.

```{r}
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ group)
```

# Filter genes with sum counts per million >= 10 across all samples
Visualize read count distribution
```{r}
raw_count_threshold <- 10
hist(log10(rowSums(counts(dds))), breaks = 50)
abline(v = log10(raw_count_threshold), col = "red", lty = 2)
```

```{r}
cpm <- apply(counts(dds),2, function(x) (x/sum(x))*1000000)
hist(log10(rowSums(cpm)))
abline(v = log10(raw_count_threshold), col = "red", lty = 2)
```

Filter to remove genes with low read counts
```{r}
keep <- rowSums(cpm) >= raw_count_threshold
dds <- dds[keep,]
dds
```

Filter to select for protein-coding genes
```{r}
transcript_type <- read_csv(file = "../01_input/biomaRt_elegans_transcript_biotype.csv")
colnames(transcript_type) <- c("WBGeneID", "genome_id", "gene_name", "biotype")
dds <- dds[rownames(dds) %in% 
      (transcript_type %>% 
         filter(biotype == "protein_coding") %>% 
         pull(WBGeneID)),]
```


Perform Differential Expression
```{r}
dds <- DESeq(dds)
resultsNames(dds)
```

# Sample-to-sample distance matrix

```{r}
vsd <- vst(dds, blind = FALSE)
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- vsd$names
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
mega_cor_plot <- pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
mega_cor_plot
```
```{r}
myPDFplot(plot = mega_cor_plot, name = "FACS_Correlation_All_Samples", height = 4.5, width = 6, plotdir = "../03_output/plots/Correlation_Matrix/")

```

# Figure S2
```{r}
all_sorted_samples_cor <- vsd.corr.per.stage("GFPplus|GFPminus", "Correlation of FACS isolated GFP+ and GFP- samples")
myPDFplot(plot = all_sorted_samples_cor, name = "All_Stage_FACS_Correlation_Sorted_Samples", height = 4, width = 6, plotdir = "../03_output/plots/Correlation_Matrix/")
```

# Remove L1 rep 2

```{r}
remove_samples <- c("L1_whole_rep2", "L1_cells_rep2", "L1_GFPplus_rep2", "L1_GFPminus_rep2")
coldata <- coldata %>% filter(!names %in% remove_samples)
dds <- dds[,!colnames(dds)%in% remove_samples]
```

# Perform Differential Expression
```{r}
dds <- DESeq(dds)
resultsNames(dds)
```

# Counts table output for supplemental datatable

```{r}
# raw counts
write.table(as.data.frame(counts(dds, normalized = FALSE)) %>% rownames_to_column(var = "WBGeneID"), 
            file = "../03_output/count_tables_for_sup/intestine_FACS_RNAseq_raw_counts.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)
# normalized counts
write.table(as.data.frame(counts(dds, normalized = TRUE)) %>% rownames_to_column(var = "WBGeneID"), 
            file = "../03_output/count_tables_for_sup/intestine_FACS_RNAseq_norm_counts.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)

# rlog transformed counts
write.table(as.data.frame(assay(rlog(dds, blind=FALSE))) %>% rownames_to_column(var = "WBGeneID"), 
            file = "../03_output/count_tables_for_sup/intestine_FACS_RNAseq_rlog_counts.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)

```


# Within-stage pairwise comparisons

## Set cutoff values
```{r}
thresh = 1
sig = 0.01
```

## Embryo stage pairwise comparisons
```{r}
embryo_alt_hyp_res_df <- alt_hyp_res_df("embryo", thresh = thresh, sig = sig)
```

```{r}
de_category_MA_plot(embryo_alt_hyp_res_df, paste("Embryo differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
```{r}
de_category_bar_plot(embryo_alt_hyp_res_df, paste("Embryo differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
## L1 stage pairwise comparisons
```{r}
L1_alt_hyp_res_df<- alt_hyp_res_df("L1", thresh = thresh, sig = sig)
```

```{r}
de_category_MA_plot(L1_alt_hyp_res_df, paste("L1 differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
```{r}
de_category_bar_plot(L1_alt_hyp_res_df, paste("L1 differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
## L3 stage pairwise comparisons

```{r}
L3_alt_hyp_res_df<- alt_hyp_res_df("L3", thresh = thresh, sig = sig)
```

```{r}
de_category_MA_plot(L3_alt_hyp_res_df, paste("L3 differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
```{r}
de_category_bar_plot(L3_alt_hyp_res_df, paste("L3 differentially expressed genes\nlfc = ",thresh," & padj < ",sig, sep = ""))
```
# Figure S3

Whole worm vs. dissociated cells analysis

```{r fig.width=5.5, fig.height=3.5}
cells_vs_whole_bar_df <- L1_alt_hyp_res_df %>% filter(isDE == TRUE, label == "whole_vs_cells") %>% group_by(type) %>% summarize(genes = n()) %>% mutate(stage = "L1") %>%
  bind_rows(
    L3_alt_hyp_res_df %>% filter(isDE == TRUE, label == "whole_vs_cells") %>% group_by(type) %>% summarize(genes = n()) %>% mutate(stage = "L3")
    ) %>%
  bind_rows(
    embryo_alt_hyp_res_df %>% filter(isDE == TRUE, label == "whole_vs_cells") %>% group_by(type) %>% summarize(genes = n()) %>% mutate(stage = "embryo")
  ) 
cells_vs_whole_bar_df

cells_vs_whole_bar_plot <- cells_vs_whole_bar_df %>%
  ggplot(aes(x = type, y = genes, fill = stage, label = genes)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7), width = 0.5, color = "black") +
  geom_text(hjust = -0.25, position = position_dodge(width = 0.7)) +
  scale_fill_manual(values = c("black", "grey", "white")) +
  theme_classic() +
  coord_flip()
cells_vs_whole_bar_plot
ggsave(plot = cells_vs_whole_bar_plot, filename = "../03_output/plots/cells_vs_whole/cells_vs_whole_bar_plot.pdf", width = 5.5, height = 3.5)  
```

```{r, fig.width=6, fig.height=3}
cells_vs_whole_MA_plot <- de_category_MA_plot(
L1_alt_hyp_res_df %>% filter(label == "whole_vs_cells") %>% mutate(label = comparison) %>%
  bind_rows(
    L3_alt_hyp_res_df %>% filter(label == "whole_vs_cells") %>% mutate(label = comparison)
    ) %>%
  bind_rows(
    embryo_alt_hyp_res_df %>% filter(label == "whole_vs_cells") %>% mutate(label = comparison)
  ),
title = NULL
)
cells_vs_whole_MA_plot
ggsave(plot = cells_vs_whole_MA_plot, filename = "../03_output/plots/cells_vs_whole/cells_vs_whole_MA_plot.pdf", width = 6, height = 3)
```

# DESeq2 builtin plotting function
```{r}
res_embryoGFPplus_vs_embryoGFPminus <- results(dds, contrast = c("group", "embryoGFPplus", "embryoGFPminus"))
res_L1GFPplus_vs_L1_GFPminus <- results(dds, contrast = c("group", "L1GFPplus", "L1GFPminus"))
res_L3GFPplus_vs_L3_GFPminus <- results(dds, contrast = c("group", "L3GFPplus", "L3GFPminus"))
```

```{r}
res_embryoGFPplus_vs_embryoGFPminus_ashr <- lfcShrink(dds, contrast = c("group", "embryoGFPplus", "embryoGFPminus"), type = "ashr")
res_L1GFPplus_vs_L1GFPminus_ashr <- lfcShrink(dds, contrast = c("group", "L1GFPplus", "L1GFPminus"), type = "ashr")
res_L3GFPplus_vs_L3GFPminus_ashr <- lfcShrink(dds, contrast = c("group", "L3GFPplus", "L3GFPminus"), type = "ashr")
```

# Export rlog counts

```{r}
all_samples_rld <- rlog(dds)
write_rds(all_samples_rld, file = "../03_output/rlog_counts/all_samples_rlog_counts.rds")
write_tsv(as.data.frame(assay(all_samples_rld)) %>% rownames_to_column(var = "WBGeneID"), file = "../03_output/rlog_counts/all_samples_rlog_counts.tsv")
write_tsv(as.data.frame(assay(all_samples_rld)) %>% rownames_to_column(var = "WBGeneID") %>% select(WBGeneID, contains("GFPplus")), file = "../03_output/rlog_counts/GFPplus_samples_rlog_counts.tsv")
all_samples_rld <- read_rds(file = "../03_output/rlog_counts/all_samples_rlog_counts.rds")
all_samples_rld_df <- read_tsv(file = "../03_output/rlog_counts/all_samples_rlog_counts.tsv")

head(all_samples_rld_df)
```

# Export intestine enrichment assignment

```{r}
embryo_intestine_gene_cats <- embryo_alt_hyp_res_df %>% 
  drop_na(padj) %>%
  filter(label == "GFPplus_vs_GFPminus", padj < sig) %>% 
  select(WBGeneID, altHyp = "type") %>% 
  mutate(intestine_expression = case_when(
    altHyp == "greater" ~ "enriched",
    altHyp == "less" ~ "depleted",
    altHyp == "lessAbs" ~ "equal")) %>% 
  mutate(intestine_expression = fct_relevel(intestine_expression, c("enriched", "equal", "depleted")))


L1_intestine_gene_cats <- L1_alt_hyp_res_df %>% 
  filter(label == "GFPplus_vs_GFPminus", padj < sig) %>% 
  select(WBGeneID, altHyp = "type") %>% 
  mutate(intestine_expression = case_when(
    altHyp == "greater" ~ "enriched",
    altHyp == "less" ~ "depleted",
    altHyp == "lessAbs" ~ "equal"))%>% 
  mutate(intestine_expression = fct_relevel(intestine_expression, c("enriched", "equal", "depleted")))

L3_intestine_gene_cats <- L3_alt_hyp_res_df %>% 
  filter(label == "GFPplus_vs_GFPminus", padj < sig) %>% 
  select(WBGeneID, altHyp = "type") %>% 
  mutate(intestine_expression = case_when(
    altHyp == "greater" ~ "enriched",
    altHyp == "less" ~ "depleted",
    altHyp == "lessAbs" ~ "equal"))%>% 
  mutate(intestine_expression = fct_relevel(intestine_expression, c("enriched", "equal", "depleted")))

write_csv(x = embryo_intestine_gene_cats, file = "../03_output/intestine_gene_categories/embryo_intestine_gene_categories.csv")

write_csv(x = L1_intestine_gene_cats, file = "../03_output/intestine_gene_categories/L1_intestine_gene_categories.csv")

write_csv(x = L3_intestine_gene_cats, file = "../03_output/intestine_gene_categories/L3_intestine_gene_categories.csv")
```


# Average embryo GFP+ sample reads

Make function
```{r}
thresh = 1
sig = 0.01

rlog_status <- function(stage, res, hyp_df){
all_samples_rld_df %>% 
  select(WBGeneID, contains(paste(stage,"GFPplus", sep = "_"))) %>% 
  pivot_longer(cols = contains("GFPplus"), values_to = "rlog_counts") %>%
  separate(name, sep = "_", into = c("stage", "sample", "rep")) %>%
  group_by(WBGeneID) %>%
  summarise(mean.rlog.counts = mean(rlog_counts), var.rlog.counts = var(rlog_counts)) %>%
  left_join(hyp_df %>% filter(label == "GFPplus_vs_GFPminus") %>% select(WBGeneID, type, isDE), by = "WBGeneID")
}
```


```{r}
embryo_rlog_status_df <- rlog_status(stage = "embryo", res = res_embryoGFPplus_vs_embryoGFPminus, hyp_df = embryo_alt_hyp_res_df)
head(embryo_rlog_status_df)
write_csv(embryo_rlog_status_df, file = "../03_output/embryo_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
embryo_rlog_status_df <- read_csv(file = "../03_output/embryo_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
```
```{r}
L1_rlog_status_df <- rlog_status(stage = "L1", res = res_L1GFPplus_vs_L1_GFPminus, hyp_df = L1_alt_hyp_res_df)
write_csv(L1_rlog_status_df, file = "../03_output/L1_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
L1_rlog_status_df <- read_csv(file = "../03_output/L1_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
head(L1_rlog_status_df)
```
```{r}
L3_rlog_status_df <- rlog_status(stage = "L3", res = res_L3GFPplus_vs_L3_GFPminus, hyp_df = L3_alt_hyp_res_df)
write_csv(L3_rlog_status_df, file = "../03_output/L3_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
L3_rlog_status_df <- read_csv(file = "../03_output/L3_GFPplus_rlog_counts_status_df.csv", col_names = TRUE)
head(L3_rlog_status_df)
```
# Intesinte expression across development

# UpSet Plot
```{r}
intestine_enriched_genes <- data.frame(embryo_rlog_status_df, stage = "embryo") %>% bind_rows(data.frame(L1_rlog_status_df, stage = "L1"), data.frame(L3_rlog_status_df, stage = "L3")) %>%filter(type == "greater", isDE == TRUE) %>% select(WBGeneID, stage)

stage_list<- list(embryo = filter(intestine_enriched_genes, stage == "embryo")$WBGeneID,
     L1 = filter(intestine_enriched_genes, stage == "L1")$WBGeneID,
     L3 = filter(intestine_enriched_genes, stage == "L3")$WBGeneID)
comb_mat <-make_comb_mat(stage_list)
UpSet(comb_mat)
```
```{r}
comb_size(comb_mat)
```

# Data output

```{r}
# per stage GFP+ vs GFP- differential expression
write_csv(res_to_df(res_embryoGFPplus_vs_embryoGFPminus), file = "../03_output/pairwise_DE_results/res_embryoGFPplus_vs_embryoGFPminus.csv")
write_csv(res_to_df(res_L1GFPplus_vs_L1_GFPminus), file = "../03_output/pairwise_DE_results/res_L1GFPplus_vs_L1GFPminus.csv")
write_csv(res_to_df(res_L3GFPplus_vs_L3_GFPminus), file = "../03_output/pairwise_DE_results/res_L3GFPplus_vs_L3GFPminus.csv")

# per stage GFP+ vs GFP- differential expression with log2FC shrink (visualization, ranking)
write_csv(res_to_df(res_embryoGFPplus_vs_embryoGFPminus_ashr), file = "../03_output/pairwise_shrunk_DE_results/res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk.csv")
write_csv(res_to_df(res_L1GFPplus_vs_L1GFPminus_ashr), file = "../03_output/pairwise_shrunk_DE_results/res_L1GFPplus_vs_L1GFPminus_ashr_shrunk.csv")
write_csv(res_to_df(res_L3GFPplus_vs_L3GFPminus_ashr), file = "../03_output/pairwise_shrunk_DE_results/res_L3GFPplus_vs_L3GFPminus_ashr_shrunk.csv")

# per stage enrichment annotation
write_csv(embryo_alt_hyp_res_df %>% filter(label == "GFPplus_vs_GFPminus") %>% select(WBGeneID:type, isDE), file = "../03_output/embryo_GFPplus_vs_GFPminus_alt_hyp_res.csv")
write_csv(L1_alt_hyp_res_df %>% filter(label == "GFPplus_vs_GFPminus") %>% select(WBGeneID:type, isDE), file = "../03_output/L1_GFPplus_vs_GFPminus_alt_hyp_res.csv")
write_csv(L3_alt_hyp_res_df %>% filter(label == "GFPplus_vs_GFPminus") %>% select(WBGeneID:type, isDE), file = "../03_output/L3_GFPplus_vs_GFPminus_alt_hyp_res.csv")

```

# Plot output

```{r}
res_embryoGFPplus_vs_embryoGFPminus_ashr <- read_csv(file = "../03_output/pairwise_shrunk_DE_results/res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk.csv")
embryo_intestine_gene_cats <- read_csv(file = "../03_output/intestine_gene_categories/embryo_intestine_gene_categories.csv")
res_embryoGFPplus_vs_embryoGFPminus_ashr %>% left_join(embryo_intestine_gene_cats, by = "WBGeneID") %>% drop_na(intestine_expression) %>% filter(intestine_expression == "enriched") %>%
  ggplot(aes(x = log10(baseMean), y =log2FoldChange)) +
  geom_point(data = res_embryoGFPplus_vs_embryoGFPminus_ashr %>% filter(!(WBGeneID %in% embryo_intestine_gene_cats$WBGeneID)),
             shape = 16, alpha = 0.5, stroke = 0, size = 1, color = "grey") +
  geom_point(shape = 16, alpha = 0.5, stroke = 0, size = 1, aes(color = intestine_expression)) +
  theme_bw() +
  labs(x = "log10(mean RNA abundance)",
       y = "log2(embryo GFP+ reads/embryo GFP- reads)")
ggsave(filename = "../03_output/plots/Intestine_Expression_Category/L3_GFPplus_vs_GFPminus_shrunk_MAplot_enriched_only.jpg", width = 5, height = 3)
  
res_to_df(res_embryoGFPplus_vs_embryoGFPminus_ashr) %>% left_join(embryo_intestine_gene_cats, by = "WBGeneID") %>% drop_na(intestine_expression) %>% group_by(intestine_expression) %>% summarise(genes = n()) %>% ggplot(aes(x = intestine_expression, y = genes, fill = intestine_expression, label = genes)) +
  geom_bar(stat = "identity") +
  geom_text(vjust = -0.25) +
  theme_bw()

```
```{r}
single_MA_plot <- function(in_res, in_cats){
  res_to_df(in_res) %>% left_join(in_cats, by = "WBGeneID") %>% drop_na(intestine_expression) %>%
    filter(log2FoldChange > -20, intestine_expression == "enriched") %>%
  ggplot(aes(x =  log10(baseMean), y =log2FoldChange)) +
  geom_point(data = res_to_df(in_res) %>% filter(!(WBGeneID %in% in_cats$WBGeneID)),
             shape = 16, alpha = 0.5, stroke = 0, size = 1, color = "grey") +
  geom_point(shape = 16, alpha = 0.5, stroke = 0, size = 1, aes(color = intestine_expression)) +
  theme_bw() +
  xlim(0.5,5.5) +
  ylim(-6, 10) +
  labs(x = "log10(mean RNA abundance)",
       y = "log2(GFP+ reads/GFP- reads)",
       title = paste("data: ", deparse(substitute(in_res)), sep = ""))
}

single_cat_bar <- function(in_res, in_cats){
  res_to_df(in_res) %>% left_join(in_cats, by = "WBGeneID") %>% drop_na(intestine_expression) %>% group_by(intestine_expression) %>% summarise(genes = n()) %>% ggplot(aes(x = intestine_expression, y = genes, fill = intestine_expression, label = genes)) +
  geom_bar(stat = "identity") +
  geom_text(vjust = -0.25) +
  ggtitle(paste("data: ", deparse(substitute(in_cats)), sep = "")) +
  theme_bw()
}
```

```{r}
embryo_shrunk_MA <- single_MA_plot(res_embryoGFPplus_vs_embryoGFPminus_ashr, embryo_intestine_gene_cats)
embryo_shrunk_MA
ggsave(plot = embryo_shrunk_MA, file = "../03_output/plots/Intestine_Expression_Category/embryo_GFPplus_vs_GFPminus_shrunk_MAplot.pdf", width = 5, height = 3)
```


```{r}
embryo_category_bar <- single_cat_bar(res_embryoGFPplus_vs_embryoGFPminus_ashr, embryo_intestine_gene_cats)
embryo_category_bar
ggsave(plot = embryo_category_bar, file = "../03_output/plots/Intestine_Expression_Category/embryo_GFPplus_vs_GFPminus_Expression_Category_barplot.pdf", width = 5, height = 3)
```

```{r}
L1_shrunk_MA <- single_MA_plot(res_L1GFPplus_vs_L1GFPminus_ashr, L1_intestine_gene_cats)
L1_shrunk_MA
ggsave(plot = L1_shrunk_MA, file = "../03_output/plots/Intestine_Expression_Category/L1_GFPplus_vs_GFPminus_shrunk_MAplot.pdf", width = 5, height = 3)

L1_category_bar <- single_cat_bar(res_L1GFPplus_vs_L1GFPminus_ashr, L1_intestine_gene_cats)
L1_category_bar
ggsave(plot = L1_category_bar, file = "../03_output/plots/Intestine_Expression_Category/L1_GFPplus_vs_GFPminus_Expression_Category_barplot.pdf", width = 5, height = 3)
```

```{r}
L3_shrunk_MA <- single_MA_plot(res_L3GFPplus_vs_L3GFPminus_ashr, L3_intestine_gene_cats)
L3_shrunk_MA
ggsave(plot = L3_shrunk_MA, file = "../03_output/plots/Intestine_Expression_Category/L3_GFPplus_vs_GFPminus_shrunk_MAplot.pdf", width = 5, height = 3)

L3_category_bar <- single_cat_bar(res_L3GFPplus_vs_L3GFPminus_ashr, L3_intestine_gene_cats)
L3_category_bar
ggsave(plot = L3_category_bar, file = "../03_output/plots/Intestine_Expression_Category/L3_GFPplus_vs_GFPminus_Expression_Category_barplot.pdf", width = 5, height = 3)
```

# Alternative Hypothesis output

```{r}
thresh <- 1
sig <- 0.01

res_embryoGFP_alHyp_greater <- results(dds, contrast = c("group", "embryoGFPplus", "embryoGFPminus"), lfcThreshold=thresh, altHypothesis = "greater", alpha = sig)
write_csv(x = res_to_df(res_embryoGFP_alHyp_greater), file = "../03_output/res_embryoGFP_alHyp_greater.csv")

res_L1GFP_alHyp_greater <- results(dds, contrast = c("group", "L1GFPplus", "L1GFPminus"), lfcThreshold=thresh, altHypothesis = "greater", alpha = sig)
write_csv(x = res_to_df(res_L1GFP_alHyp_greater), file = "../03_output/res_L1GFP_alHyp_greater.csv")

res_L3GFP_alHyp_greater <- results(dds, contrast = c("group", "L3GFPplus", "L3GFPminus"), lfcThreshold=thresh, altHypothesis = "greater", alpha = sig)
write_csv(x = res_to_df(res_L3GFP_alHyp_greater), file = "../03_output/res_L3GFP_alHyp_greater.csv")


```

# Session info

```{r}
sessionInfo()
```
