---
title: "novel_intestine_genes"
author: "Robert Williams"
date: "4/5/2022"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load packages

```{r}
library(tidyverse)
```

Load data

```{r}
embryo_intestine_gene_categories <- read_csv(file = "../03_output/intestine_gene_categories/embryo_intestine_gene_categories.csv")
L1_intestine_gene_categories <- read_csv(file = "../03_output/intestine_gene_categories/L1_intestine_gene_categories.csv")
L3_intestine_gene_categories <- read_csv(file = "../03_output/intestine_gene_categories/L3_intestine_gene_categories.csv")

res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk <- read_csv("../03_output/pairwise_shrunk_DE_results/res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk.csv")
res_L1GFPplus_vs_L1GFPminus_ashr_shrunk <- read_csv("../03_output/pairwise_shrunk_DE_results/res_L1GFPplus_vs_L1GFPminus_ashr_shrunk.csv")
res_L3GFPplus_vs_L3GFPminus_ashr_shrunk <- read_csv("../03_output/pairwise_shrunk_DE_results/res_L3GFPplus_vs_L3GFPminus_ashr_shrunk.csv")

intestine_genes <- read_delim("../../01_tissue_specific_genes/01_input/genes_direct_and_inferred_for_WBbt_0005772_intestine.txt", delim = "\t",col_names = c("WBGeneID", "gene_name", "Species"))
```
# Embryo genes

```{r}
embryo_enriched_annotation <- embryo_intestine_gene_categories %>% 
  filter(altHyp == "greater") %>% 
  left_join(res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk, by = "WBGeneID") %>% 
  left_join(intestine_genes, by = "WBGeneID") %>%
  mutate(annotation = case_when(is.na(gene_name) ~ "unannotated",
                                !is.na(gene_name) ~ "annotated"))

embryo_enriched_annotation %>% filter(annotation == "unannotated") %>% slice_min(padj, n = 10)
```
Smallest p-values:

1. WBGene00001940, his-66, Predicted to enable DNA binding activity and protein heterodimerization activity. Predicted to be part of nucleosome. Is an ortholog of several human genes including H2BC11 (H2B clustered histone 11); H2BC13 (H2B clustered histone 13); and H2BC9 (H2B clustered histone 9).
2. WBGene00043097, C02D5.4, Predicted to enable glutathione dehydrogenase (ascorbate) activity. Predicted to be involved in cellular oxidant detoxification and glutathione metabolic process.
3. WBGene00018992, trpl-1, pseudogene
4. WBGene00022014, Y61A9LA.1, Predicted to enable transmembrane transporter activity. Predicted to be involved in transmembrane transport.
5. WBGene00020522, T15B7.10 pseudogene
6. WBGene00009106, acs-12, Predicted to enable CoA-ligase activity. 
7. WBGene00021611, nhr-238, Predicted to be involved in regulation of transcription by RNA polymerase II.
8. WBGene00001922, his-48, Predicted to enable DNA binding activity. Involved in innate immune response. Predicted to be located in chromosome and nucleus. Predicted to be part of nucleosome. Expressed in gonad
9. WBGene00000509, cka-1, Enables choline kinase activity. Involved in CDP-choline pathway. Located in cell cortex.
10. WBGene00017327, F10C1.8, Is predicted to encode a protein with the following domains: Phosphorylation site; Lamin Tail Domain; Lamin tail domain; and Lamin tail domain superfamily.


# L1 genes

```{r}
L1_enriched_annotation <- L1_intestine_gene_categories %>% 
  filter(altHyp == "greater") %>% 
  left_join(res_L1GFPplus_vs_L1GFPminus_ashr_shrunk, by = "WBGeneID") %>% 
  left_join(intestine_genes, by = "WBGeneID") %>%
  mutate(annotation = case_when(is.na(gene_name) ~ "unannotated",
                                !is.na(gene_name) ~ "annotated"))

L1_enriched_annotation %>% filter(annotation == "unannotated") %>% slice_min(padj, n = 10)
```
smallest p-values:  

1. WBGene00001940, his-66, Predicted to enable DNA binding activity and protein heterodimerization activity. Predicted to be part of nucleosome. Is an ortholog of several human genes including H2BC11 (H2B clustered histone 11); H2BC13 (H2B clustered histone 13); and H2BC9 (H2B clustered histone 9).
2. WBGene00018992, trpl-1, pseudogene
3. WBGene00017327, F10C1.8, Is predicted to encode a protein with the following domains: Phosphorylation site; Lamin Tail Domain; Lamin tail domain; and Lamin tail domain superfamily.
4. WBGene00020683, ribo-1, Predicted to be involved in protein N-linked glycosylation via asparagine. Predicted to be located in endoplasmic reticulum membrane. Predicted to be integral component of membrane
5. WBGene00012134, T28F3.8, Enriched in I5 neuron; cholinergic neurons; head mesodermal cell; and retrovesicular ganglion based on RNA-seq and microarray studies.
6. WBGene00018925, F56A11.5, Predicted to enable catalytic activity; molybdenum ion binding activity; and pyridoxal phosphate binding activity. Predicted to be located in membrane. Predicted to be integral component of membrane.
7. WBGene00235352, ZK250.13, Enriched in head mesodermal cell based on RNA-seq studies. Is affected by several genes including elt-2; eat-2; and pmk-1 
8. WBGene00017957, F31F4.1, Predicted to be located in membrane. Predicted to be integral component of membrane.
9. WBGene00002070, ile-1, Predicted to enable mannose binding activity. Involved in IRE1-mediated unfolded protein response
10. WBGene00001922, his-48, Predicted to enable DNA binding activity. Involved in innate immune response. Predicted to be located in chromosome and nucleus. Predicted to be part of nucleosome. Expressed in gonad


# L3 genes

```{r}
L3_enriched_annotation <- L3_intestine_gene_categories %>% 
  filter(altHyp == "greater") %>% 
  left_join(res_L3GFPplus_vs_L3GFPminus_ashr_shrunk, by = "WBGeneID") %>% 
  left_join(intestine_genes, by = "WBGeneID") %>%
  mutate(annotation = case_when(is.na(gene_name) ~ "unannotated",
                                !is.na(gene_name) ~ "annotated"))

L3_enriched_annotation %>% filter(annotation == "unannotated") %>% slice_min(padj, n = 10)
```
Smallest p-value:  

1. WBGene00001940, his-66, Predicted to enable DNA binding activity and protein heterodimerization activity. Predicted to be part of nucleosome. Is an ortholog of several human genes including H2BC11 (H2B clustered histone 11); H2BC13 (H2B clustered histone 13); and H2BC9 (H2B clustered histone 9).
2. WBGene00016469, C36B7.6, Acts upstream of or within IRE1-mediated unfolded protein response. Predicted to be located in endomembrane system and membrane.
3. WBGene00017327, F10C1.8, Enriched in body wall musculature and head mesodermal cell based on RNA-seq and microarray studies. Is affected by several genes including daf-16; daf-2; and dpy-10 based on microarray; tiling array; and RNA-seq studies.
4. WBGene00001922, his-48, Predicted to enable DNA binding activity. Involved in innate immune response. Predicted to be located in chromosome and nucleus. Predicted to be part of nucleosome. Expressed in gonad
5. WBGene00206354, F27E5.9, Predicted to be located in membrane. Predicted to be integral component of membrane.
6. WBGene00012134, T28F3.8, Enriched in I5 neuron; cholinergic neurons; head mesodermal cell; and retrovesicular ganglion based on RNA-seq and microarray studies. Is affected by several genes including daf-16; daf-2; and glp-1 based on microarray; RNA-seq; and proteomic studies. 
7. WBGene00017957, F31F4.1, Predicted to be located in membrane. Predicted to be integral component of membrane.
8. WBGene00017974, vps-41, Predicted to enable metal ion binding activity. Involved in negative regulation of apoptotic process; vacuolar protein processing; and vacuolar transport. Predicted to be located in late endosome. Predicted to be part of HOPS complex. 
9. WBGene00011850, T20B3.1, Predicted to enable carnitine O-octanoyltransferase activity. Predicted to be located in peroxisome.
10. WBGene00003635, nhr-45, Predicted to enable DNA-binding transcription factor activity; sequence-specific DNA binding activity; and zinc ion binding activity.

```{r}
embryo_enriched_annotation %>% filter(WBGeneID %in% unique(L1_enriched_annotation$WBGeneID, L3_enriched_annotation$WBGeneID)) %>% filter(annotation == "unannotated") %>% arrange(padj)
```
  
1. his-66
2. WBGene00043097, C02D5.4, Predicted to enable glutathione dehydrogenase (ascorbate) activity and glutathione transferase activity. Predicted to be involved in glutathione metabolic process. Predicted to be located in cytoplasm.
3. pseudogene
4. pseudogene
5. WBGene00001922, his-48, Predicted to enable DNA binding activity. Involved in innate immune response. Predicted to be located in chromosome and nucleus. Predicted to be part of nucleosome. Expressed in gonad
6. WBGene00017327, F10C1.8, Enriched in body wall musculature and head mesodermal cell based on RNA-seq and microarray studies. Lamin tail domain superfamily.
7. WBGene00001932, his-58
8. WBGene00000071, acy-4, Predicted to enable adenylate cyclase activity. Involved in oocyte growth. Acts upstream of or within with a positive effect on oocyte maturation. Acts upstream of or within positive regulation of gene expression. Predicted to be located in plasma membrane. Predicted to be integral component of plasma membrane.
9. WBGene00199158, C33D3.6, ncRNA
10. WBGene00003585, ndx-8, Enables acetyl-CoA hydrolase activity; hydroxymethylglutaryl-CoA hydrolase activity; and succinyl-CoA hydrolase activity. Involved in coenzyme A catabolic process. Predicted to be located in peroxisome.
11. WBGene00011677, cyp-13A1, Predicted to enable heme binding activity; iron ion binding activity; and oxidoreductase activity.
12. WBGene00012134, T28F3.8, Enriched in I5 neuron; cholinergic neurons; head mesodermal cell; and retrovesicular ganglion based on RNA-seq and microarray studies.
13. WBGene00000267, bre-2, Predicted to enable acetylglucosaminyltransferase activity. Involved in response to toxic substance. Predicted to be located in Golgi membrane.
14. WBGene00003978, pes-4, Predicted to enable mRNA binding activity. Predicted to be involved in regulation of RNA metabolic process and regulation of gene expression.
15. WBGene00014233, ZK1128.7, Expressed in vulval muscle
16. WBGene00021820, nipa-1, Predicted to enable magnesium ion transmembrane transporter activity. Involved in adult locomotory behavior. Located in neuronal cell body. Used to study hereditary spastic paraplegia.
17. WBGene00011756, ctg-2, Enriched in head mesodermal cell and sensory neurons based on RNA-seq studies. Is affected by several genes including daf-2; hsf-1; and let-7 based on microarray; proteomic; and RNA-seq studies.
18. WBGene00011850, T20B3.1, Predicted to enable carnitine O-octanoyltransferase activity. Predicted to be located in peroxisome.
19. WBGene00206354, F27E5.9, Predicted to be located in membrane. Predicted to be integral component of membrane.
20. WBGene00044211, Y60A3A.25, Is affected by several genes including daf-12; hsf-1; and eat-2 based on tiling array; microarray; and RNA-seq studies. 

WBGene00043097, C02D5.4
WBGene00000071, acy-4
WBGene00000267, bre-2

```{r fig.width=3, fig.height = 4}
novel_genes <- data.frame(WBGeneID = c("WBGene00043097", "WBGene00000071", "WBGene00000267"), gene_name = c("C02D5.4", "acy-4", "bre-2"))
res_emb <- read_csv("../03_output/pairwise_DE_results/res_embryoGFPplus_vs_embryoGFPminus.csv")
res_L1 <- read_csv("../03_output/pairwise_DE_results/res_L1GFPplus_vs_L1GFPminus.csv")
res_L3 <- read_csv("../03_output/pairwise_DE_results/res_L3GFPplus_vs_L3GFPminus.csv")

novel_gene_data <- res_emb %>% filter(WBGeneID %in% novel_genes$WBGeneID) %>% mutate(stage = "emb") %>%
  bind_rows(
    res_L1 %>% filter(WBGeneID %in% novel_genes$WBGeneID) %>% mutate(stage = "L1")
  ) %>%
  bind_rows(
    res_L3 %>% filter(WBGeneID %in% novel_genes$WBGeneID) %>% mutate(stage = "L3")
  ) %>%
  left_join(novel_genes, by = "WBGeneID")
head(novel_gene_data)
```


```{r fig.width=3, fig.height = 4}
novel_gene_plot <- ggplot(novel_gene_data, aes(x = stage, y = log2FoldChange, fill = stage)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_errorbar(aes(ymin=log2FoldChange-lfcSE, ymax=log2FoldChange+lfcSE), width=.2,
                 position=position_dodge(.9)) +
  scale_y_continuous(breaks = seq(0,6, by = 1)) +
  facet_grid(gene_name~.) +
  theme_classic()

novel_gene_plot
# ggsave(plot = novel_gene_plot, filename = "../03_output/plots/novel_intestine_genes/220815_Novel_Intestine_Genes_log2FC.pdf", width = 3, height = 4)
```


# Session info

```{r}
sessionInfo()
```
