---
title: "Signal Correlations and ANOVA for Supplemental figures 9 and 10"
author: "DC King - Onish lab"
date: "5/19/2022"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message=FALSE)
library(tidyr)
library(dplyr)
library(GenomicRanges)
library(magrittr)
library(knitr) # for kable
library(ggplot2)
library(cowplot)
library(multcompView) # for signal_cor_functions.R
library(ggridges)
library(ggpubr)
library(dunn.test) # for signal_cor_functions.R
library(ggpmisc)
library(rms)
#install.packages('contrast') 
library('contrast')
source('david-reader.R')
source('signal_cor_functions.R') # for lm.combo and unused HSD and dunn functions
okabe <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
```

## ELT-2 data

```{r read-data-ELT-2}
elt2.data = read_ELT2_binding_data(as_genomic_ranges = TRUE) # from david-reader.R
#glimpse(elt2.data)
```

## Rob data

```{r read-rob-merged}
# david-reader.R
rob = read_rob_all_merged() %>% dplyr::select(-starts_with("pvalue."),-starts_with("lfcSE."))
glimpse(rob)
```

```{r combine-elt-2-rnaseq}
merge = right_join(rob, elt2.data, by = "WBGeneID")
# glimpse(merge)

# Left-over genes are NA in alldata. Give them a label
merge$embryo_int_exp[is.na(merge$embryo_int_exp)] = 'ftr'
merge$L1_int_exp[is.na(merge$L1_int_exp)] = 'ftr'
merge$L3_int_exp[is.na(merge$L3_int_exp)] = 'ftr'

merge$embryo.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$LE_bound, "bound", "unbound"), merge$embryo_int_exp)
merge$L1.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L1_bound, "bound", "unbound"), merge$L1_int_exp)
merge$L3.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L3_bound, "bound", "unbound"), merge$L3_int_exp)

# fix a name 
merge %<>% dplyr::rename(L3.log_chip_signal_mean = L3_log.chip_signal_mean)
```

```{r log2-chip-to-log10, eval=FALSE}
merge %<>% dplyr::mutate(LE.log_chip_signal_mean = LE.log_chip_signal_mean*log(2,10),
                  LE.log_chip_signal_max  = LE.log_chip_signal_max*log(2,10),
                  L1.log_chip_signal_mean = L1.log_chip_signal_mean*log(2,10),
                  L1.log_chip_signal_max  = L1.log_chip_signal_max*log(2,10),
                  L3.log_chip_signal_mean = L3.log_chip_signal_mean*log(2,10),
                  L3.log_chip_signal_max  = L3.log_chip_signal_max*log(2,10)
                  )
```


```{r output-promoters}
merge %>% select(seqnames, start, end, WBGeneID, embryo_int_exp, LE_bound) %>% filter(embryo_int_exp == "enriched") %>% select(-embryo_int_exp) %>%  write.table(file="promoter_files/promoters_embryo_bound_TF.bed", sep="\t", quote=FALSE, row.names = FALSE, col.names = FALSE)
```

```{r choose-columns}

alldata = merge %>% dplyr::select(ends_with("log_chip_signal_mean"), 
                        starts_with("rlogc."),
                        starts_with("log2FoldChange."),
                        ends_with(".boundEnrichedStatus"),
                        ends_with("_int_exp"),
                        ends_with("_bound"),
                        din.status.description,
                        wikigene_name,
                        WBGeneID)

alldata$embryo_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))
alldata$L1_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))
alldata$L3_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))
```

# Linear models

### Embryo

```{r embryo-trim-data}
#data = alldata %>% filter(embryo_int_exp %in% c('enriched','equal'))
data = alldata %>% filter(embryo_int_exp %in% c('enriched'))
#data %<>% filter(WBGeneID != 'WBGene00016153') # negative signal, lowest point
#data %<>% filter(WBGeneID != 'WBGene00012762') # kind of hot region, highest point in "depleted"

# trim extreme .1%
q.LE.chipsig = quantile(data$LE.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.embryo = quantile(data$rlogc.embryo, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(LE.log_chip_signal_mean, q.LE.chipsig[1],q.LE.chipsig[2]))
data %<>% filter(between(rlogc.embryo, q.rlogc.embryo[1],q.rlogc.embryo[2]))
```

```{r embryo-lm}
# base model (0): chip signal
mod.embryo0 = lm.combo(rlogc.embryo ~ LE.log_chip_signal_mean, 
                       data = data)

mod.embryo0$label = "ChIP signal only"
mod.embryo1 = lm.combo(rlogc.embryo ~ LE.log_chip_signal_mean * LE_bound, 
                       data = data)

.mod.embryo1 = lm(rlogc.embryo ~ LE.log_chip_signal_mean * LE_bound, 
                       data = data)

embryo_contrast_bound_unbound = contrast(.mod.embryo1, list(LE_bound=FALSE,
                    LE.log_chip_signal_mean=mean(data$LE.log_chip_signal_mean)),
         list(LE_bound=TRUE,
              LE.log_chip_signal_mean=mean(data$LE.log_chip_signal_mean))
         )

mod.embryo1$label = "ChIP signal AND ELT-2 binding"
```

```{r embryo-pearson-cor}
LE.straightup.cor = cor(data$LE.log_chip_signal_mean, data$rlogc.embryo)
dbound = data %>% filter(LE_bound)
dunbound = data %>% filter(!LE_bound)
LE.bound.cor = cor.test(dbound$LE.log_chip_signal_mean, dbound$rlogc.embryo, conf.level=.975)
LE.unbound.cor = cor.test(dunbound$LE.log_chip_signal_mean, dunbound$rlogc.embryo, conf.level=.975)

```

```{r skipped-embryo-plot-models, include=FALSE}

data %<>% mutate(elt2.binding = factor(ifelse(LE_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))

p1=ggplot(data, 
       aes(x=LE.log_chip_signal_mean, 
           y=rlogc.embryo, 
           #group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  geom_smooth(color="black", method="lm", formula = y~x, se=F) + 
  #stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x) +
  ggtitle(bquote(.(mod.embryo0$label) ~ R^2 == .(round(mod.embryo0$r.squared,3)))) +
  ylab("DESeq2 rlog counts") +
  xlab("LE log ChIP promoter signal") +
  scale_color_manual(values=okabe) #+ 
  #guides(color="none")

p2=ggplot(data,
       aes(x= LE.log_chip_signal_mean, 
           y=rlogc.embryo, 
           #group=elt2.binding, 
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x, color="black") +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  xlab("LE log ChIP promoter signal") +
  ggtitle("Embryo Expression level (ELT-2 bound/unbound)") +
  scale_color_manual(values=okabe) + 
  guides(color="none") + theme_bw()

p3=ggplot(data %>% filter(!is.na(din.status.description)),
       aes(x= LE.log_chip_signal_mean, 
           y=rlogc.embryo, 
           #group=elt2.binding, 
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x) +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~din.status.description ) +
  ylab("DESeq2 rlog counts") + 
  xlab("LE log ChIP promoter signal") +
  ggtitle("Embryo Expression level (ELT-2 bound/unbound)") +
  scale_color_manual(values=okabe) + 
  guides(color="none"); p3


title <- ggdraw() + 
  draw_label(
    "Embryo: Intestine enriched and equal expressed genes predicted by ChIP signal level",
    fontface = 'bold',
    x = 0,
    hjust = 0
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
# plot_grid(title, p1, p2, p3, ncol=1,labels=c('','A','B','C'), rel_heights = c(0.1, 1, 1, 1)); 
#ggsave("plots/signal_cor.xy.embryo.pdf", height=15,width=12)

# middle one only (bound vs. unbound)
plot_grid(title, p2, ncol=1,rel_heights = c(0.1,1)); 

# replacing with all stages facetted
#ggsave("plots/signal_cor.xy.embryo.pdf", height=5,width=12)

LE.p.title = title
LE.p1 = p1
```

```{r skipped-get-embryo-set, include=FALSE}
data = alldata %>% dplyr::filter(embryo_int_exp %in% c('enriched') & LE_bound &  din.status.description %in% c("down_ELT2_minus", "up_ELT2_minus")) %>% select(WBGeneID, LE.log_chip_signal_mean, rlogc.embryo, din.status.description, embryo_int_exp, LE_bound, wikigene_name)

# trim extreme .1%
# q.LE.chipsig = quantile(data$LE.log_chip_signal_mean, c(1-.9995,.9995))
# q.rlogc.embryo = quantile(data$rlogc.embryo, c(1-.9995,.9995),na.rm=T)
# data %<>% filter(between(LE.log_chip_signal_mean, q.LE.chipsig[1],q.LE.chipsig[2]))
# data %<>% filter(between(rlogc.embryo, q.rlogc.embryo[1],q.rlogc.embryo[2]))

nrow(data)
head(data)
write.table(data, file="embryo_bound_elt2_minus_changed.tsv", sep="\t", row.names = F, quote = F)
```

### L1

This section works on data with a Dineen status description only. Later chunks that include the stage as a variable use the full set.

```{r L1-dineen-trim-data}
data = alldata %>% filter(L1_int_exp %in% c('enriched'))
#data %<>% filter(WBGeneID != 'WBGene00016153') # negative signal, lowest point
#data %<>% filter(WBGeneID != 'WBGene00012762') # kind of hot region, highest point in "depleted"
data %<>% filter(!is.na(din.status.description))


# trim extreme .1%
q.L1.chipsig = quantile(data$L1.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.L1 = quantile(data$rlogc.L1, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(L1.log_chip_signal_mean, q.L1.chipsig[1],q.L1.chipsig[2]))
data %<>% filter(between(rlogc.L1, q.rlogc.L1[1],q.rlogc.L1[2]))
```


```{r L1-dineen-lm}


# base model (0): chip signal
mod.L10 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean, 
                       data = data)

mod.L10$label = "ChIP signal only"
mod.L11 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean * L1_bound, 
                       data = data)
.mod.L11 = lm(rlogc.L1 ~ L1.log_chip_signal_mean * L1_bound, 
                       data = data)

L1_contrast_bound_unbound = contrast(.mod.L11, list(L1_bound=FALSE,
                    L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean)),
         list(L1_bound=TRUE,
              L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean))
         )

mod.L11$label = "ChIP signal AND ELT-2 binding"


mod.L12 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean * L1_bound * din.status.description, 
                       data = data)
.mod.L12 = lm(rlogc.L1 ~ L1.log_chip_signal_mean * L1_bound * din.status.description, 
                       data = data)
```

Contrasts seem very useful but I don't know if they are applicable to the big picture.

```{r skipped-L1-dineen-contrasts, include=FALSE}
contrast(.mod.L12, list(L1_bound=TRUE,
                       din.status.description="unchanged_ELT2_minus",
                       L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean)),
         list(L1_bound=TRUE,
              din.status.description="up_ELT2_minus",
              L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean))
         )
contrast(.mod.L12, list(L1_bound=FALSE,
                       din.status.description="unchanged_ELT2_minus",
                       L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean)),
         list(L1_bound=TRUE,
              din.status.description="unchanged_ELT2_minus",
              L1.log_chip_signal_mean=mean(data$L1.log_chip_signal_mean))
         )
```

Since ELT-2 binding is not a significant coefficient, how do the t-tests on the remaining coefficients change when you take it out?

```{r L1-dineen-model-remove-ELT2-dep}
# remove binding to see if an ELT-2-dependency coefficient is significant by t-test
data$din.status.description=factor(data$din.status.description, levels=c("unchanged_ELT2_minus", "up_ELT2_minus","down_ELT2_minus" ))
mod.L13 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean * din.status.description, 
                       data = data)
mod.L13
```

Skipped to focus on R-squared and coefficient estimates/p-values.

``` {r skipped-L1-dineen-pearson-cor, include=FALSE}
L1.straightup.cor = cor(data$L1.log_chip_signal_mean, data$rlogc.L1)
dbound = data %>% filter(L1_bound)
dunbound = data %>% filter(!L1_bound)
L1.bound.cor = cor.test(dbound$L1.log_chip_signal_mean, dbound$rlogc.L1, conf.level=.975)
L1.unbound.cor = cor.test(dunbound$L1.log_chip_signal_mean, dunbound$rlogc.L1, conf.level=.975)
```

```{r L1-dineen-data-rename-L1_bound}
data %<>% mutate(elt2.binding = factor(ifelse(L1_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))
```

Skipped because it is included in the final plots.

```{r skipped-L1-dineen-plots, include=FALSE}
p1=ggplot(data, 
       aes(x=L1.log_chip_signal_mean, 
           y=rlogc.L1, 
           #group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  geom_smooth(color="black", method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x) +
  ggtitle(bquote(.(mod.L10$label) ~ R^2 == .(round(mod.L10$r.squared,3))),
          subtitle = "Dineen data only") +
  ylab("DESeq2 rlog counts") +
  xlab("L1 log ChIP promoter signal") +
  scale_color_manual(values=okabe);p1

p2=ggplot(data,
       aes(x= L1.log_chip_signal_mean, 
           y=rlogc.L1, 
           #group=elt2.binding, 
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x, color="black") +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  xlab("L1 log ChIP promoter signal") +
  ggtitle("L1 Expression level (ELT-2 bound/unbound)", subtitle="Dineen data only") +
  scale_color_manual(values=okabe) + 
  guides(color="none") + theme_bw(); p2

data.L1 = data
```
### Supplemental Figure 10b

```{r L1-signal-cor-plot}
# relevel to down, unchanged, up, for display. having the unchanged first is
# useful for categorical regression
levels(data.L1$din.status.description) = c("down_ELT2_minus", 
                                           "unchanged_ELT2_minus", 
                                           "up_ELT2_minus")
data = data.L1
p3=ggplot(data %>% filter(!is.na(din.status.description)),
       aes(x= L1.log_chip_signal_mean, 
           y=rlogc.L1,
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(method="lm", formula = y~x, se=F, color="black", fullrange = T) + 
  stat_poly_eq(parse=T, 
               aes(label = paste(after_stat(..rr.label..), 
                                 after_stat(..n.label..), 
                                 sep="*\", \"*")), # https://www.rdocumentation.org/packages/ggpmisc/versions/0.4.5/topics/stat_poly_eq
          
               formula=y~x, 
               color="black",
               label.y=0.01) +
  geom_density_2d(colour="black", 
                  size=.1, 
                  contour_var = "ndensity",
                  alpha=1) +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~L1_bound + din.status.description,
             labeller = 
               labeller(.cols=function(variable,value){
               l = list( down_ELT2_minus = "down",
                         unchanged_ELT2_minus = "unchanged",
                         up_ELT2_minus = "up")
               return(l[value])})) +
  ylab("DESeq2 rlog counts") + 
  xlab("L1 log ChIP promoter signal") +
  ggtitle("B) ELT-2 responsive genes have the same relationship to ChIP signal in peaks versus not") +
  scale_color_manual(values=okabe) + 
  theme_bw() +
  theme(
        strip.background = element_blank(),
        strip.text = element_text(size=10),
    plot.title = element_text(face="bold")
  ) +
  guides(color="none"); p3
pdf(file="plots/signal_cor.xy.L1.dineen.status.pdf", height=5,width=8, family="ArialMT")
p3
dev.off()
#ggsave("plots/signal_cor.xy.L1.dineen.status.pdf", height=5,width=12)
```
### Supplemental Figure 10a

```{r violin}
#levels(data.L1$din.status.description) <- c("down_ELT2_minus"  ,"unchanged_ELT2_minus", "up_ELT2_minus")
down.t = t.test(rlogc.L1 ~ elt2.binding, data = data.L1 %>% filter(!is.na(din.status.description) & din.status.description == "down_ELT2_minus"), conf.level = 1-(.01/3) )

up.t = t.test(rlogc.L1 ~ elt2.binding, data = data.L1 %>% filter(!is.na(din.status.description) & din.status.description == "up_ELT2_minus"), conf.level = 1-(.01/3) )

unchanged.t = t.test(rlogc.L1 ~ elt2.binding, data = data.L1 %>% filter(!is.na(din.status.description) & din.status.description == "unchanged_ELT2_minus"), conf.level = 1-(.01/3) )


up.t$q.value = min(up.t$p.value*3,1)
unchanged.t$q.value = min(unchanged.t$p.value*3,1)
down.t$q.value = min(down.t$p.value*3,1)

# this part is bullshit because geom_text(parse=T) reformats the %.2e into %.f
up.formatted.exp = floor(log(up.t$q.value,10))
down.formatted.exp = floor(log(down.t$q.value,10))
up.formatted.coef = round( up.t$q.value * 10 ^ -up.formatted.exp, 1)
down.formatted.coef = round( down.t$q.value * 10 ^ -down.formatted.exp, 1)


fmt_string = "italic(p) == %.2f%%*%%10^%d" # %%*%% resolves to %*% in sprintf, which resolve to a small times character (x) in geom_text(parse=T), which is plotmath

annotation_text = data.frame(elt2.binding=1.5,
                             rlogc.L1 = c(21,21,22),
                             lab = c(
                               sprintf(fmt_string, down.formatted.coef,
                                       down.formatted.exp),
                               sprintf("italic(p) == %.2f", unchanged.t$q.value),
                               sprintf(fmt_string, up.formatted.coef,
                                       up.formatted.exp)
                               ),
                             din.status.description = factor(c("down_ELT2_minus", "unchanged_ELT2_minus", "up_ELT2_minus"))
                             )

p = ggviolin(data.L1 %>% filter(!is.na(din.status.description)),
     x= "elt2.binding", 
     y="rlogc.L1", 
     add = c("mean_ci"),
     add.params = list(size=.1,
                       width=.75,
                       color="black",
                       ci=1 - (.01/3)),
    color="elt2.binding",
    error.plot = "crossbar",
    fill="elt2.binding",
    alpha=.5,
    palette = okabe,
    ggtheme=theme_bw(),
    facet.by = "din.status.description"
    ,panel.labs=list(din.status.description=c("unchanged", "up", "down")) # i don't know what the hell is going on here, but the order is really screwed up
) + 
  ggtitle("A) Higher mean expression of ELT-2 bound, ELT-2-dependent promoters") +
  xlab("") + ylab("Expression level (rlog counts)") +
  theme(
    axis.text.x = element_text(#angle = 90, 
                           colour="black",vjust=.5),
    axis.text.y=element_text(colour="black"),
    strip.background = element_blank(),
    plot.title = element_text(face="bold"),
    strip.text = element_text(size=12)) +
  guides(fill="none",color="none") + 
  geom_text(data = annotation_text, 
            mapping = aes(x=elt2.binding,
                          y=rlogc.L1,
                          label=lab),
            parse=T) +
    geom_line(data= rbind(
              data.frame(x=c(1,1,2,2),
                            y=c(20,21,21,20)-1,
                            din.status.description="down_ELT2_minus"),
              data.frame(x=c(1,1,2,2),
                            y=c(20,21,21,20)-1,
                            din.status.description="unchanged_ELT2_minus"),
              data.frame(x=c(1,1,2,2),
                            y=c(19,20,20,19)+1,
                            din.status.description="up_ELT2_minus")
              ),
            mapping=aes(x=x,y=y))
p
pdf(file="plots/signal_cor.violin.L1.dineen.status.pdf", height=3.2,width=8, family="ArialMT")
p
dev.off()

```


```{r skipped-L1-ggdraw, include=FALSE}
title <- ggdraw() + 
  draw_label(
    "L1: Intestine enriched and equal expressed genes predicted by ChIP signal level",
    fontface = 'bold',
    x = 0,
    hjust = 0
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
# plot_grid(title, p1, p2, p3, ncol=1,labels=c('','A','B','C'), rel_heights = c(0.1, 1, 1, 1));  ggsave("plots/signal_cor.xy.L1.pdf", height=15,width=12)

# middle one only (bound vs. unbound)
plot_grid(title, p2, ncol=1,rel_heights = c(0.1,1)); 
#ggsave("plots/signal_cor.xy.L1.pdf", height=5,width=12)

L1.p.title = title
L1.p1 = p1


# based on the ANOVA, use a reduced model to isolate meaningful coefficients for meaningful contrasts
data$din.status.description = factor(data$din.status.description, c("unchanged_ELT2_minus", "down_ELT2_minus", "up_ELT2_minus"))
mod.reduced0 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean * L1_bound * din.status.description, data = data)
mod.reduced = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean:L1_bound + din.status.description, data = data)
smack = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean + din.status.description, data = data)
smack2 = lm.combo(rlogc.L1 ~ L1.log_chip_signal_mean*L1_bound  + din.status.description, data = data)

```

## L3

```{r L3-filter-trim-data}
data = alldata %>% filter(L3_int_exp %in% c('enriched'))
#data %<>% filter(WBGeneID != 'WBGene00016153') # negative signal, lowest point
#data %<>% filter(WBGeneID != 'WBGene00012762') # kind of hot region, highest point in "depleted"

# trim extreme .1%
q.L3.chipsig = quantile(data$L3.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.L3 = quantile(data$rlogc.L3, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(L3.log_chip_signal_mean, q.L3.chipsig[1],q.L3.chipsig[2]))
data %<>% filter(between(rlogc.L3, q.rlogc.L3[1],q.rlogc.L3[2]))
```

```{r L3-lm}
# base model (0): chip signal
mod.L30 = lm.combo(rlogc.L3 ~ L3.log_chip_signal_mean, 
                       data = data)

mod.L30$label = "ChIP signal only"
mod.L31 = lm.combo(rlogc.L3 ~ L3.log_chip_signal_mean * L3_bound, 
                       data = data)
.mod.L31 = lm(rlogc.L3 ~ L3.log_chip_signal_mean * L3_bound, 
                       data = data)

L3_contrast_bound_unbound = contrast(.mod.L31, list(L3_bound=FALSE,
                    L3.log_chip_signal_mean=mean(data$L3.log_chip_signal_mean)),
         list(L3_bound=TRUE,
              L3.log_chip_signal_mean=mean(data$L3.log_chip_signal_mean))
         )

mod.L31$label = "ChIP signal AND ELT-2 binding"

L3.straightup.cor = cor(data$L3.log_chip_signal_mean, data$rlogc.L3)
dbound = data %>% filter(L3_bound)
dunbound = data %>% filter(!L3_bound)
L3.bound.cor = cor.test(dbound$L3.log_chip_signal_mean, dbound$rlogc.L3, conf.level=.975)
L3.unbound.cor = cor.test(dunbound$L3.log_chip_signal_mean, dunbound$rlogc.L3, conf.level=.975)
```

```{r L3-rename-L3_bound}
data %<>% mutate(elt2.binding = factor(ifelse(L3_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))
```

```{r skipped-L3-plot, include=FALSE}
p1=ggplot(data, 
       aes(x=L3.log_chip_signal_mean, 
           y=rlogc.L3, 
           #group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  geom_smooth(color="black", method="lm", formula = y~x, se=F) + 
  #stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x) +
  #ggtitle(bquote(.(mod.L30$label) ~ R^2 == .(round(mod.L30$r.squared,3)))) +
  ylab("DESeq2 rlog counts") +
  xlab("L3 log ChIP promoter signal") +
  scale_color_manual(values=okabe) #+ 
  #guides(color="none")
p1

p2=ggplot(data,
       aes(x= L3.log_chip_signal_mean, 
           y=rlogc.L3, 
           #group=elt2.binding, 
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x, color="black") +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  xlab("L3 log ChIP promoter signal") +
  ggtitle("L3 Expression level (ELT-2 bound/unbound)") +
  scale_color_manual(values=okabe) + 
  guides(color="none") + theme_bw()

p3=ggplot(data %>% filter(!is.na(din.status.description)),
       aes(x= L3.log_chip_signal_mean, 
           y=rlogc.L3, 
           #group=elt2.binding, 
           color=elt2.binding
           )) + geom_point() +
  stat_smooth(method="lm", formula = y~x, se=F) + 
  stat_poly_eq(parse=T, aes(label = ..rr.label..), formula=y~x) +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~din.status.description ) +
  ylab("DESeq2 rlog counts") + 
  xlab("L3 log ChIP promoter signal") +
  ggtitle("L3 Expression level (ELT-2 bound/unbound)") +
  scale_color_manual(values=okabe) + 
  guides(color="none"); p3


title <- ggdraw() + 
  draw_label(
    "L3: Intestine enriched and equal expressed genes predicted by ChIP signal level",
    fontface = 'bold',
    x = 0,
    hjust = 0
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
# plot_grid(title, p1, p2, p3, ncol=1,labels=c('','A','B','C'), rel_heights = c(0.1, 1, 1, 1));  ggsave("plots/signal_cor.xy.L3.pdf", height=15,width=12)

# middle one only (bound vs. unbound)
plot_grid(title, p2, ncol=1,rel_heights = c(0.1,1)); 
#ggsave("plots/signal_cor.xy.L3.pdf", height=5,width=12)

L3.p.title = title
L3.p1 = p1
```

```{r skipped-all-plots, include=FALSE}
title <- ggdraw() + 
  draw_label(
    "Expression level of Intestine enriched and equal expressed genes predicted by ChIP signal intensity",
    fontface = 'bold',
    x = 0,
    hjust = 0
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )

plot_grid(title, 
          LE.p1 + ggtitle(bquote("Embryo " ~ R^2 == .(round(mod.embryo0$r.squared,3)))),
          L1.p1 + ggtitle(bquote("L1 " ~ R^2 == .(round(mod.L10$r.squared,3)))),
          L3.p1 + ggtitle(bquote("L3 " ~ R^2 == .(round(mod.L30$r.squared,3)))),
          ncol=1,labels=c('','A','B','C'), rel_heights = c(0.1, 1, 1, 1))
#ggsave("plots/signal_cor.xy.final.pdf", height=15,width=12)
```

```{r kabels-regression-stats}

kable(mod.embryo1$coefficients, digits=4)
kable(mod.L11$coefficients, digits=4)
kable(mod.L12$anova, digits=4)
kable(mod.L31$coefficients, digits=4)

write.table(mod.embryo1$coefficients, file="../03_output/correlation_regression.stats.embryo.txt", sep="\t", quote=F)
write.table(mod.embryo1$anova, file="../03_output/correlation_regression.anova.embryo.txt", sep="\t", quote=F)
write.table(mod.L11$coefficients, file="../03_output/correlation_regression.stats.L1.txt", sep="\t", quote=F)
write.table(mod.L12$anova, file="../03_output/correlation_regression.anova.dineen.L1.txt", sep="\t", quote=F)
write.table(mod.L31$coefficients, file="../03_output/correlation_regression.stats.L3.txt", sep="\t", quote=F)
write.table(mod.L31$anova, file="../03_output/correlation_regression.anova.L3.txt", sep="\t", quote=F)

# use "contrast" to test bound vs not (lm coefficients only test the intercept and slope terms individually)

cat(capture.output(embryo_contrast_bound_unbound), file="../03_output/correlation_regression.contrast_bound_unbound.embryo.txt", sep="\t")

cat(capture.output(L1_contrast_bound_unbound), file="../03_output/correlation_regression.contrast_bound_unbound.L1.txt", sep="\t")

cat(capture.output(L3_contrast_bound_unbound), file="../03_output/correlation_regression.contrast_bound_unbound.L3.txt", sep="\t")

```

# All data by stage- supplemental figure 9

```{r format-all-data}

data = alldata %>% filter(embryo_int_exp == "enriched") %>% 
  select(LE.log_chip_signal_mean, LE_bound, rlogc.embryo) %>%
  dplyr::rename(log_chip_signal_mean=LE.log_chip_signal_mean,
                rlogc = rlogc.embryo,
                ELT2bound = LE_bound) %>%
  mutate(stage = "embryo")
data = alldata %>% filter(L1_int_exp == "enriched") %>% 
  select(L1.log_chip_signal_mean, L1_bound, rlogc.L1) %>%
  dplyr::rename(log_chip_signal_mean=L1.log_chip_signal_mean,
                rlogc = rlogc.L1,
                ELT2bound = L1_bound) %>%
        mutate(stage = "L1")  %>%
        rbind(data)
                                                                                    
data = alldata %>% filter(L3_int_exp == "enriched") %>% 
  select(L3.log_chip_signal_mean, L3_bound, rlogc.L3) %>%
  dplyr::rename(log_chip_signal_mean=L3.log_chip_signal_mean,
                rlogc = rlogc.L3,
                ELT2bound = L3_bound) %>%
        mutate(stage = "L3")  %>%
        rbind(data)
```

## Supplemental Figure 9a

```{r t-tests-for-violins}
embryo.t = t.test(rlogc ~ ELT2bound, data = data %>% filter(stage == "embryo"), 
       conf.level = 1-(.01/3) )
L1.t = t.test(rlogc ~ ELT2bound, data = data %>% filter(stage == "L1"), 
       conf.level = 1-(.01/3) )
L3.t = t.test(rlogc ~ ELT2bound, data = data %>% filter(stage == "L3"), 
       conf.level = 1-(.01/3) )

embryo.t$q.value = min(1, embryo.t$p.value*3)
L1.t$q.value = min(1, L1.t$p.value*3)
L3.t$q.value = min(1, L3.t$p.value*3)

```

```{r label-formatting-for-scientific-notation}
# this part is bullshit but it will match the x10^exp in the L1 section
embryo.formatted.exp = floor(log(embryo.t$q.value,10))
embryo.formatted.coef = round( embryo.t$q.value * 10 ^ -embryo.formatted.exp, 1)

L1.formatted.exp = floor(log(L1.t$q.value,10))
L1.formatted.coef = round( L1.t$q.value * 10 ^ -L1.formatted.exp, 1)

L3.formatted.exp = floor(log(L3.t$q.value,10))
L3.formatted.coef = round( L3.t$q.value * 10 ^ -L3.formatted.exp, 1)

fmt_string = "italic(p) == %.2f%%*%%10^%d" # %%*%% resolves to %*% in sprintf, which resolve to a small times character (x) in geom_text(parse=T), which is plotmath
```

```{r annotation-text-dataframe}
annotation_text = data.frame(ELT2bound = FALSE,
                             rlogc = c(22,22,21),
                             lab = c(sprintf(fmt_string, 
                                             embryo.formatted.coef,
                                             embryo.formatted.exp),
                                   sprintf(fmt_string, 
                                           L1.formatted.coef,
                                           L1.formatted.exp),
                                   sprintf(fmt_string, 
                                           L3.formatted.coef,
                                           L3.formatted.exp)),
                             stage = factor(c("embryo", "L1", "L3"))
                             )
```

```{r violin-plot}
p=ggviolin(data=data, x="ELT2bound",y="rlogc",   
        add = c("mean_ci"),
#        draw_quantiles = c(.5),
     add.params = list(size=.1,
                         width=.75,
                       color="black",
                       ci=1 - (.01/3)),

    color="ELT2bound",
    error.plot = "crossbar",
    fill="ELT2bound",
    alpha=.5,
    palette = okabe,
    ggtheme=theme_bw(),
    facet.by = "stage") + 
  ggtitle("A) Higher mean expression level of intestine-enriched genes in ELT-2 bound promoters") +
  xlab("") + ylab("Expression level (rlog counts)") +
#  theme_classic() +
  theme(axis.text.x=element_text(colour="black", #angle=90,
                                 vjust=.5),
        axis.text.y=element_text(colour="black"),
        strip.background = element_blank(),
        strip.text = element_text(size=10),
        plot.title = element_text(face="bold")) +
  guides(fill="none",color="none") + 
  scale_x_discrete(labels=c("ELT-2 unbound", "ELT-2 bound")) +
    geom_text(data = annotation_text, 
            mapping = aes(x=1.5,
                          y=rlogc,
                          label=lab),
                          parse=TRUE
                          ) +
  geom_line(data= rbind(
              data.frame(x=c(1,1,2,2),
                            y=c(20,21,21,20),
                            stage="embryo"),
              data.frame(x=c(1,1,2,2),
                            y=c(20,21,21,20),
                            stage="L1"),
              data.frame(x=c(1,1,2,2),
                            y=c(19,20,20,19),
                            stage="L3")
              ),
            mapping=aes(x=x,y=y));p

pdf(file="plots/signal_cor.violin.stages.pdf", height=4,width=8, family="ArialMT")
p
dev.off()         
```

### FWER on the coefficient t-tests

```{r fwer-on-stage-coefficients}
data$stage = factor(data$stage, levels=c("L1", "embryo","L3"))
mod.log_chip_mean = lm.combo(rlogc ~ log_chip_signal_mean, data=data)
mod.log_chip_mean = lm.combo(rlogc ~ log_chip_signal_mean, data=data)
mod.all = lm.combo(rlogc ~ log_chip_signal_mean * stage * ELT2bound, data=data)
mod.notStage = lm.combo(rlogc ~ log_chip_signal_mean * ELT2bound, data=data)
mod.notELT2bound = lm.combo(rlogc ~ log_chip_signal_mean * stage, data=data)
# FWER
fwer = cbind(mod.all$coefficients[,c(1,4)], 
             p.adjust(mod.all$coefficients[,4],method = "holm"),
             p.adjust(mod.all$coefficients[,4],method = "bonferroni"))
colnames(fwer) <- c("Estimate", "Pr(>|t|)", "FWER (holm)", "FWER (bonferroni)")
fwer
```

## Supplemental Figure 9b
```{r stages-extract-r-squared-from-models}
emb.r2 = mod.embryo1$r.squared
L1.r2 = mod.L11$r.squared
L3.r2 = mod.L31$r.squared
```

```{r stages-signal-cor-plot}
p2=ggplot(data %>% mutate(stage=forcats::fct_relevel(stage, c("embryo","L1","L3"))),
       aes(x= log_chip_signal_mean, 
           y=rlogc, 
           group=ELT2bound,
           color=ELT2bound
           )) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x, se=F, fullrange = T) + 
  stat_poly_eq(parse=T, 
              aes(label = paste(after_stat(..rr.label..), 
                 after_stat(..n.label..), 
                 sep="*\", \"*")),
               formula=y~x, 
               color="black",
               label.y=0.02) +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~ELT2bound + stage) +
  #facet_wrap(~stage ) +
  ylab("DESeq2 rlog counts") + 
  xlab("All log ChIP promoter signal") +
  ggtitle("B) Intestine-enriched genes have the same relationship to ChIP signal in peaks versus not") +
  scale_color_manual(values=okabe) +  
  geom_density_2d(colour="black", 
                  size=.1, 
                  contour_var = "ndensity") +
  guides(color="none") + theme_bw() + theme(
        strip.background = element_blank(),
        strip.text = element_text(size=10),
    plot.title = element_text(face="bold"))
p2
pdf(file="plots/signal_cor.xy.stage.final.pdf",height=5,width=7.7, family="ArialMT")
p2
dev.off()
#ggsave("plots/signal_cor.xy.stage.final.pdf", height=8,width=12)                           
```

End of visualizations and analyses used in the paper.

# Other visualizations/analyses not used

### filled density plots

```{r filled-density-plots, include=FALSE}
ggplot(data %>% mutate(stage=forcats::fct_relevel(stage, c("embryo","L1","L3"))),
       aes(x= log_chip_signal_mean, 
           y=rlogc, 
           group=ELT2bound,
           color=ELT2bound
           )) + 
  
  stat_smooth(color="black", method="lm", formula = y~x, se=F, fullrange = T) + 
  stat_poly_eq(parse=T, 
               aes(label = paste(after_stat(..rr.label..), 
                                 after_stat(..n.label..), 
                                 sep="*\", \"*")), # https://www.rdocumentation.org/packages/ggpmisc/versions/0.4.5/topics/stat_poly_eq
               formula=y~x, 
               color="black",
               label.y=0) +
  ylim(c(0,19)) + 
  #xlim(c(0,11)) +
  facet_wrap(~ELT2bound + stage ) +
  #facet_wrap(~stage ) +
  ylab("DESeq2 rlog counts") + 
  xlab("All log ChIP promoter signal") +
  ggtitle("B) Low gene expression variance explained by ChIP promoter signal") +
  scale_color_manual(values=okabe) +  
  geom_point(size=.1, color="white") +
  geom_density_2d_filled(
                  size=.1,
                  contour_var = "ndensity") +
  
  guides(color="none") + theme_bw() + 
  theme(text = element_text(family = "Arial", face = "bold"))
```

### using datadist (rms) and contrast (rms)

```{r contrasts-all-model, include=FALSE}

.mod.all = lm(rlogc ~ log_chip_signal_mean * stage * ELT2bound, data=data)

d = datadist(data$rlogc, data$log_chip_signal_mean)
options(datadist="d") 


ols.mod = ols(rlogc ~ log_chip_signal_mean * stage * ELT2bound, data=data)

chip_sigs = data %>%filter(stage=="embryo") %>% pull(log_chip_signal_mean) %>% range()

contrast_embryo_binding=contrast(ols.mod, list(ELT2bound=FALSE,
                    log_chip_signal_mean=chip_sigs,
                    stage = "embryo"),
                  list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage="embryo"), conf.int = 1-(.05/3))
df_contrast_embryo_binding = contrast_embryo_binding[c("log_chip_signal_mean", "stage", "Contrast","Lower","Upper")] %>% as.data.frame()



chip_sigs = data %>%filter(stage=="L1") %>% pull(log_chip_signal_mean) %>% range()
contrast_L1_binding=contrast(ols.mod, list(ELT2bound=FALSE,
                    log_chip_signal_mean=chip_sigs,
                    stage = "L1"),
                  list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage="L1"), conf.int = 1-(.05/3))
df_contrast_L1_binding = contrast_L1_binding[c("log_chip_signal_mean", "stage",  "Contrast","Lower","Upper")] %>% as.data.frame()




chip_sigs = data %>%filter(stage=="L3") %>% pull(log_chip_signal_mean) %>% range()
contrast_L3_binding = contrast(ols.mod, list(ELT2bound=FALSE,
                    log_chip_signal_mean=chip_sigs,
                    stage = "L3"),
                  list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage="L3"), conf.int = 1-(.05/3))
df_contrast_L3_binding = contrast_L3_binding[c("log_chip_signal_mean","stage", "Contrast","Lower","Upper")] %>% as.data.frame()

df_binding_contrasts = rbind(
  df_contrast_embryo_binding,
  df_contrast_L1_binding,
  df_contrast_L3_binding
)

ggplot(df_binding_contrasts, aes(x=log_chip_signal_mean, y=Contrast)) + geom_line() + 
   geom_ribbon(aes(ymin=Lower, ymax=Upper), width=0, alpha=0.15, linetype=0) +
  facet_wrap(~stage) + theme_bw()


chip_sigs = data %>% filter(stage %in% c("L1","L3")) %>% pull(log_chip_signal_mean) %>% range()
contrast_embryo_L1 = contrast(ols.mod, list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage = c("embryo")),
         list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage = c( "L1")))
df_contrast_embryo_L1 = contrast_embryo_L1[c("log_chip_signal_mean","Contrast","Lower","Upper")] %>% as.data.frame()

chip_sigs = data %>% filter(stage %in% c("L1","L3")) %>% pull(log_chip_signal_mean) %>% range()
contrast_L1_L3 = contrast(ols.mod, list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage = c("L1")),
         list(ELT2bound=TRUE,
                    log_chip_signal_mean=chip_sigs,
                    stage = c( "L3")));#contrast_L1_L3
df_c_L1_L3 = contrast_L1_L3[c("log_chip_signal_mean","Contrast","Lower","Upper")] %>% as.data.frame()

ggplot(df_c_L1_L3, aes(x=log_chip_signal_mean, y=Contrast)) + geom_line() + 
   geom_ribbon(aes(ymin=Lower, ymax=Upper), width=0, alpha=0.15, linetype=0)
```

```{r construct-data-rsquare-check, include=FALSE}

d.X = seq(8,10,length.out=150)
d.Y = 1.1 * d.X + rnorm(length(d.X))
df = data.frame(d.X,d.Y,slope=1.1,sd=1)

d.X = seq(8,10,length.out=150)
d.Y = 1.1 * d.X + rnorm(length(d.X), 0, 2)
df = rbind(df, data.frame(d.X,d.Y,slope=1.1,sd=2))


d.X = seq(8,10,length.out=150)
d.Y = 1.01 * d.X + rnorm(length(d.X))
df = rbind(df, data.frame(d.X,d.Y,slope=1.01,sd=1))

d.X = seq(8,10,length.out=150)
d.Y = 1.01 * d.X + rnorm(length(d.X), 0, 2)
df = rbind(df, data.frame(d.X,d.Y,slope=1.01,sd=2))

df$sd = factor(df$sd)
df$slope = factor(df$slope)
ggplot(df, aes(x=d.X,y=d.Y, color=slope)) + geom_point() + 
  stat_smooth(color="black", method="lm", formula = y~x, se=F, fullrange = T) +
  stat_poly_eq(parse=T, 
               aes(label = paste(after_stat(..rr.label..), 
                                 after_stat(..n.label..), 
                                 sep="*\", \"*")), # https://www.rdocumentation.org/packages/ggpmisc/versions/0.4.5/topics/stat_poly_eq
               formula=y~x, 
               color="black",
               label.y=0) +
  geom_density_2d(colour="black", size=.1, alpha=1) + 
  ylim(c(0,19)) + 
  facet_wrap(~sd + slope) + 
  xlim(c(5,14)) + 
  scale_color_manual(values=okabe) +
  guides(color="none") + theme_bw()
```
