---
title: "expression_heatmaps"
author: "Rtpw"
date: "3/31/2022"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Goal: make heatmaps that will be used to visualize intestine gene expression or ELT-2 transcriptional dependence for each stage to compliment deeptools plots generated by David.

The heatmaps will be broken out into the following categories:  

- ELT-2 bound, intestine enriched  
- ELT-2 bound, intestine non-enriched  
- ELT-2 not bound, intestine enriched  
- ELT-2 not bound, intestine non-enriched  

the rows within each category will be arranged from high to low for elt-2 (-) log2FoldChange.

# Load libraries

```{r}
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
source("../../elt2_paper_functions.R")
```



# Analysis and plotting function

```{r}

expression_heatmaps <- function(stage, chip.df, gut_categories, gut_res, elt2_res){
heatmap_df <- chip.df %>% 
  mutate(binding_status = case_when(is.na(IDR_mean) ~ "not_bound", !is.na(IDR_mean) ~ "bound")) %>%
  select(WBGeneID, gene_name, binding_status) %>%
  left_join(gut_categories %>% select(WBGeneID, intestine_expression), by = "WBGeneID") %>% 
  mutate(intestine_expression = case_when(intestine_expression == "enriched" ~ "enriched",
                                          intestine_expression != "enriched" ~ "not_enriched",
                                          is.na(intestine_expression) ~ "not_enriched")) %>%
  left_join(res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk %>% 
              select(WBGeneID, gutFC = log2FoldChange, gutPadj = padj), by = "WBGeneID"
            ) %>%
  left_join(res_elt2D_v_wt_ashr_shrunk %>% select(WBGeneID, elt2FC = log2FoldChange),
            by = "WBGeneID"
            ) %>%
  replace_na(list(gutFC = 0, elt2FC = 0)) %>%
  mutate(row_split = case_when(
    binding_status == "bound" & intestine_expression == "enriched" ~"classA",
    binding_status == "bound" & intestine_expression == "not_enriched" ~"classB",
    binding_status == "not_bound" & intestine_expression == "enriched" ~"classC",
    binding_status == "not_bound" & intestine_expression == "not_enriched" ~"classD"
    )) %>%
  arrange(desc(elt2FC))

heatmap_mat <- heatmap_df %>% 
  select(WBGeneID, gutFC, elt2FC) %>% 
  column_to_rownames(var = "WBGeneID") %>% 
  as.matrix()

gut_FC <- Heatmap(name = paste(stage,"_gut_FC", sep = ""),
        heatmap_mat[,"gutFC"],
        show_row_names = FALSE,
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        row_title_rot = 0,
        col = colorRamp2(c(min(heatmap_df$gutFC), 0,max(heatmap_df$gutFC)), c("blue", "white","darkgoldenrod1")),
        row_split = heatmap_df$row_split,
        width = unit(0.5, "in"),
        border_gp = gpar(col = "black", lty = 1))

elt2_FC <- Heatmap(name = "elt2_FC",
        heatmap_mat[,"elt2FC"],
        show_row_names = FALSE,
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        row_title_rot = 0,
        col = colorRamp2(c(min(heatmap_df$elt2FC),0,max(heatmap_df$elt2FC)), c("darkgreen","white", "deeppink")),
        row_split = heatmap_df$row_split,
        width = unit(0.5, "in"),
        border_gp = gpar(col = "black", lty = 1))
#darkorchid4
#deeppink
gut_FC + elt2_FC
}


```
# Embryo Stage

Load data

```{r message=FALSE}
embryo_intestine_gene_categories<- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/intestine_gene_categories/embryo_intestine_gene_categories.csv")
res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/pairwise_shrunk_DE_results/res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk.csv")
elt2_regulated_gene_sets <- read_csv("../../03_elt2_RNAseq/03_output/elt2_regulated_gene_sets.csv")
res_elt2D_v_wt_ashr_shrunk <- read_csv("../../03_elt2_RNAseq/03_output/res_elt2D_v_wt_ashr_shrunk.csv")
LE.promoters.hilo <- read.table("../../04_promoters/03_output/LE.promoters.hilo.tsv") %>% rownames_to_column(var = "WBGeneID")

res_embryoGFP_alHyp_greater <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/res_embryoGFP_alHyp_greater.csv")
```

```{r}
embryo_expression_heatmaps <- expression_heatmaps("LE",LE.promoters.hilo, embryo_intestine_gene_categories, res_embryoGFPplus_vs_embryoGFPminus_ashr_shrunk, res_elt2D_v_wt_ashr_shrunk)
embryo_expression_heatmaps


myPDFplot(embryo_expression_heatmaps, name = "embryo_expression_heatmaps", height = 4, width = 3, plotdir = "../03_output/")
```
# L1 stage
```{r message=FALSE}
L1_intestine_gene_categories<- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/intestine_gene_categories/L1_intestine_gene_categories.csv")
res_L1GFPplus_vs_L1GFPminus_ashr_shrunk <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/pairwise_shrunk_DE_results/res_L1GFPplus_vs_L1GFPminus_ashr_shrunk.csv")
L1.promoters.hilo <- read.table("../../04_promoters/03_output/L1.promoters.hilo.tsv") %>% rownames_to_column(var = "WBGeneID")

res_L1GFP_alHyp_greater <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/res_L1GFP_alHyp_greater.csv")
```

```{r}
L1_expression_heatmaps <- expression_heatmaps("L1",L1.promoters.hilo, L1_intestine_gene_categories, res_L1GFPplus_vs_L1GFPminus_ashr_shrunk, res_elt2D_v_wt_ashr_shrunk)
L1_expression_heatmaps
myPDFplot(L1_expression_heatmaps, name = "L1_expression_heatmaps", height = 4, width = 3, plotdir = "../03_output/")
```
# L3 stage
```{r message=FALSE}
L3_intestine_gene_categories<- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/intestine_gene_categories/L3_intestine_gene_categories.csv")
res_L3GFPplus_vs_L3GFPminus_ashr_shrunk <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/pairwise_shrunk_DE_results/res_L3GFPplus_vs_L3GFPminus_ashr_shrunk.csv")
L3.promoters.hilo <- read.table("../../04_promoters/03_output/L3.promoters.hilo.tsv") %>% rownames_to_column(var = "WBGeneID")

res_L3GFP_alHyp_greater <- read_csv("../../02_emb_L1_L3_intestine_RNAseq/03_output/res_L3GFP_alHyp_greater.csv")
```

```{r}
L3_expression_heatmaps <- expression_heatmaps("L3",L3.promoters.hilo, L3_intestine_gene_categories, res_L3GFPplus_vs_L3GFPminus_ashr_shrunk, res_elt2D_v_wt_ashr_shrunk)
L3_expression_heatmaps 
myPDFplot(L3_expression_heatmaps, name = "L3_expression_heatmaps", height = 4, width = 3, plotdir = "../03_output/")
```


# Session info

```{r}
sessionInfo()
```
